bigquery-writer/bigquery-writer.py

```diff
@@ -1,38 +1,60 @@
 #!/usr/bin/env python
 import sys
 import os
-from datetime import datetime
-sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), ".."))
-from redis import Redis
-from listenstore.redis_pubsub import RedisPubSubSubscriber, RedisPubSubPublisher, NoSubscriberNameSetException, WriteFailException
 import ujson
 import logging
-from listen import Listen
+import pika
+from datetime import datetime
 from time import time, sleep
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), ".."))
+from listen import Listen
 import config
+from redis import Redis
+from redis_keys import UNIQUE_QUEUE_SIZE_KEY
 
 from googleapiclient import discovery
 from googleapiclient.errors import HttpError
 from oauth2client.client import GoogleCredentials
 
 REPORT_FREQUENCY = 5000
-SUBSCRIBER_NAME = "bq"
-KEYSPACE_NAME_INCOMING = "ilisten"
-KEYSPACE_NAME_UNIQUE = "ulisten"
 APP_CREDENTIALS_FILE = os.environ.get('GOOGLE_APPLICATION_CREDENTIALS')
 
 # TODO:
 #   Big query hardcoded data set ids
 
-class BigQueryWriterSubscriber(RedisPubSubSubscriber):
-    def __init__(self, redis):
-        RedisPubSubSubscriber.__init__(self, redis, KEYSPACE_NAME_UNIQUE, __name__)
+class BigQueryWriter(object):
+    def __init__(self):
+        self.log = logging.getLogger(__name__)
+        logging.basicConfig()
+        self.log.setLevel(logging.INFO)
+
+        self.redis = None
+        self.connection = None
+        self.channel = None
 
         self.total_inserts = 0
         self.inserts = 0
         self.time = 0
 
-    def write(self, listens):
+    def connect_to_rabbitmq(self):
+        while True:
+            try:
+                self.connection = pika.BlockingConnection(pika.ConnectionParameters(host=config.RABBITMQ_HOST, port=config.RABBITMQ_PORT))
+                break
+            except Exception as e:
+                self.log.error("Cannot connect to rabbitmq: %s, sleeping 2 seconds" % str(e))
+                sleep(2)
+
+
+    @staticmethod
+    def static_callback(ch, method, properties, body, obj):
```

Comment:
Assuming it's the one in the `start` below, it can be modified like this:
```python
self.channel.basic_consume(
    lambda ch, method, properties, body:
        self.static_callback(self, ch, method, properties, body),
    queue='unique'
)
```